ingestion/src/metadata/ingestion/source/dashboard/qlikcloud/metadata.py

```diff
@@ -0,0 +1,340 @@
+                    right=CreateChartRequest(
+                        name=chart.qInfo.qId,
+                        displayName=chart.qMeta.title,
+                        description=chart.qMeta.description,
+                        chartType=ChartType.Other,
+                        sourceUrl=chart_url,
+                        service=self.context.dashboard_service,
+                    )
+                )
+            except Exception as exc:  # pylint: disable=broad-except
+                yield Either(
+                    left=StackTraceError(
+                        name=dashboard_details.qDocName,
+                        error=f"Error creating chart [{chart}]: {exc}",
+                        stackTrace=traceback.format_exc(),
+                    )
+                )
+
+    def yield_datamodel(self, _: QlikApp) -> Iterable[Either[DashboardDataModel]]:
+        if self.source_config.includeDataModels:
+            self.data_models = self.client.get_dashboard_models()
+            for data_model in self.data_models or []:
+                try:
+                    data_model_name = (
+                        data_model.tableName if data_model.tableName else data_model.id
+                    )
+                    if filter_by_datamodel(
+                        self.source_config.dataModelFilterPattern, data_model_name
+                    ):
+                        self.status.filter(data_model_name, "Data model filtered out.")
+                        continue
+
+                    data_model_request = CreateDashboardDataModelRequest(
+                        name=data_model.id,
+                        displayName=data_model_name,
+                        service=self.context.dashboard_service,
+                        dataModelType=DataModelType.QlikCloudDataModel.value,
+                        serviceType=DashboardServiceType.QlikCloud.value,
+                        columns=self.get_column_info(data_model),
+                    )
+                    yield Either(right=data_model_request)
+                    self.register_record_datamodel(datamodel_request=data_model_request)
+                except Exception as exc:
+                    name = (
+                        data_model.tableName if data_model.tableName else data_model.id
+                    )
+                    yield Either(
+                        left=StackTraceError(
+                            name=name,
+                            error=f"Error yielding Data Model [{name}]: {exc}",
+                            stackTrace=traceback.format_exc(),
+                        )
+                    )
+
+    def get_column_info(self, data_source: QlikApp) -> Optional[List[Column]]:
+        """Build data model columns"""
+        datasource_columns = []
+        for field in data_source.fields or []:
+            try:
+                parsed_fields = {
+                    "dataTypeDisplay": "Qlik Field",
+                    "dataType": DataType.UNKNOWN,
```

Comment:
can't we do better on the data types?

Reply:
the websocket response of data models does not contain any info about data type of column, that's why we are going with unknown data type.